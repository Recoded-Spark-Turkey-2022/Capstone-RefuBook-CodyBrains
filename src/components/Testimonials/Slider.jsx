import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import testimonialsData from '../../data/testimonialsData';

// The Slider component is a pagination component that uses SwiperJS to display slides of content.
const Slider = () => {
  return (
    // The Swiper component is wrapped around the content that we want to display as slides.
    // We pass in the pagination prop and set clickable to true to enable pagination for the Swiper.
    // We also pass in the Pagination module to enable pagination functionality.
    <Swiper
      pagination={{
        clickable: true,
      }}
      modules={[Pagination]}
      className="flex items-center justify-center w-full gap-3"
      loop
    >
      {/* We map over the testimonialsData array to render a SwiperSlide for each item in the array. */}
      {testimonialsData.map((item) => {
        return (
          // The SwiperSlide component is used to wrap the content that we want to display as a slide.
          // We pass a unique key prop to each SwiperSlide to help with rendering performance.
          <SwiperSlide key={item.id} className=" flex flex-col  gap-3 ">
            {/* The content for each slide is a div containing an image, a title, and some text. */}

            <h3 className="text-lg text-refubookBlack font-normal md:text-justify  text-center">
              {item.comment}
            </h3>
            <span className="text-md  text-refubookLightBlue font-light md:text-left text-center">
              {item.name}
            </span>
            <span className="text-md  text-refubookLightBlue font-light md:text-left text-center mb-8">
              {item.country}
            </span>
          </SwiperSlide>
        );
      })}
    </Swiper>
  );
};

export default Slider;
