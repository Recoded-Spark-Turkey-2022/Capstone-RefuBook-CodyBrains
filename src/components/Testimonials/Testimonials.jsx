import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import { useTranslation } from 'react-i18next';

import map from '../../assets/world-map.svg';

const Testimonials = () => {
  const { t } = useTranslation();
  const testimonialsData = [
    {
      id: 1,
      name: t('Maria'),
      comment: t('Maria_Comment'),
      country:t('Maria_Country'),
    },
    {
      id: 1,
      name: t('Hassan'),
      comment: t('Hassan_Comment'),
      country:t('Hassan_Country'),
    },
    {
      id: 1,
      name: t('Olga'),
      comment: t('Olga_Comment'),
      country:t('Olga_Country'),
    },
    {
      id: 1,
      name: t('John'),
      comment: t('John_Comment'),
      country:t('John_Country'),
    },
  ]

  return (
    <div className="flex flex-col md:flex-row items-center justify-evenly w-full bg-refubookWhite px-8 py-4">
      <div className="flex flex-col items-center justify-center max-w-md">
        <img src={map} alt="" className="w-96 h-96" />
      </div>
      <div className="flex flex-col items-center justify-center max-w-xs md:max-w-md px-4">
        <Swiper
          pagination={{
            clickable: true,
          }}
          modules={[Pagination]}
          className="flex items-center justify-center w-full gap-3"
          loop
        >
          {/* We map over the testimonialsData array to render a SwiperSlide for each item in the array. */}
          {testimonialsData.map((item) => {
            return (
              // The SwiperSlide component is used to wrap the content that we want to display as a slide.
              // We pass a unique key prop to each SwiperSlide to help with rendering performance.
              <SwiperSlide key={item.id} className=" flex flex-col  gap-3 ">
                {/* The content for each slide is a div containing an image, a title, and some text. */}

                <h3 className="text-lg text-refubookBlack font-normal md:text-justify  text-center">
                  {item.comment}
                </h3>
                <span className="text-md  text-refubookLightBlue font-light md:text-left text-center">
                  {item.name}
                </span>
                <span className="text-md  text-refubookLightBlue font-light md:text-left text-center mb-8">
                  {item.country}
                </span>
              </SwiperSlide>
            );
          })}
        </Swiper>
      </div>
    </div>
  );
};

export default Testimonials;
