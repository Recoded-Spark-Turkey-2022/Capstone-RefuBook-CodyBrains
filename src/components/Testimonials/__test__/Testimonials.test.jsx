import React from 'react';
import { render } from '@testing-library/react';
import Testimonials from '../Testimonials';

// Mocking Swiper and SwiperSlide to avoid errors in tests

const mockedSwiper = jest.fn();
const mockedSwiperSlide = jest.fn();
const mockedPagination = jest.fn();

jest.mock('swiper/react', () => ({
  ...jest.requireActual('swiper/react'),
  Swiper: mockedSwiper,
  SwiperSlide: mockedSwiperSlide,
}));

jest.mock('swiper', () => ({
  ...jest.requireActual('swiper'),
  Pagination: mockedPagination,
}));
// Mocking testimonialsData to avoid errors in tests and to make sure the snapshot is consistent
jest.mock('../../../data/testimonialsData', () => [
  {
    id: 1,
    name: 'John Doe',
    country: 'United States',
    comment:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl.',
  },
  {
    id: 2,
    name: 'Jane Doe',
    country: 'United Kingdom',
    comment:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl.',
  },
]);

describe('Testimonials', () => {
  it('renders correctly', () => {
    const { container } = render(<Testimonials />);
    expect(container).toMatchSnapshot();
  });
  it('renders testimonialsData correctly', () => {
    const { getByText } = render(<Testimonials />);
    expect(getByText('John Doe')).toBeInTheDocument();
    expect(getByText('Jane Doe')).toBeInTheDocument();
  });
});
