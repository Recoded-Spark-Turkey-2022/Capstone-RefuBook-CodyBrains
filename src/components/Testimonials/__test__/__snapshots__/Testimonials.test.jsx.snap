// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials renders correctly 1`] = `
<div
  className="flex flex-col md:flex-row items-center justify-evenly w-full bg-refubookWhite px-8 py-4"
>
  <div
    className="flex flex-col items-center justify-center max-w-md"
  >
    <img
      alt=""
      className="w-96 h-96"
      src="world-map.svg"
    />
  </div>
  <div
    className="flex flex-col items-center justify-center max-w-xs md:max-w-md px-4"
  >
    <Swiper
      className="flex items-center justify-center w-full gap-3"
      loop={true}
      modules={
        Array [
          "Pagination",
        ]
      }
      pagination={
        Object {
          "clickable": true,
        }
      }
    >
      <SwiperSlide
        className=" flex flex-col  gap-3 "
      >
        <h3
          className="text-lg text-refubookBlack font-normal md:text-justify  text-center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl.
        </h3>
        <span
          className="text-md  text-refubookLightBlue font-light md:text-left text-center"
        >
          John Doe
        </span>
        <span
          className="text-md  text-refubookLightBlue font-light md:text-left text-center mb-8"
        >
          United States
        </span>
      </SwiperSlide>
      <SwiperSlide
        className=" flex flex-col  gap-3 "
      >
        <h3
          className="text-lg text-refubookBlack font-normal md:text-justify  text-center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl. Sed euismod, nunc vel tincidunt lacinia, nisl nisl aliquam nisl, eu aliquam nunc nisl euismod nisl.
        </h3>
        <span
          className="text-md  text-refubookLightBlue font-light md:text-left text-center"
        >
          Jane Doe
        </span>
        <span
          className="text-md  text-refubookLightBlue font-light md:text-left text-center mb-8"
        >
          United Kingdom
        </span>
      </SwiperSlide>
    </Swiper>
  </div>
</div>
`;
